src/driver.mli

```diff
@@ -173,6 +182,39 @@ val register_correction : loc:Location.t -> repl:string -> unit
 (** Hook called before processing a file *)
 val register_process_file_hook : (unit -> unit) -> unit
 
+module V2 : sig
+  (** Same as [Driver.register_transformation], but the callbacks [?impl], [?intf],
+      [?preprocess_impl], [?preprocess_intf], [?lint_impl] and [?lint_intf] have access
+      to an expansion context. Their signatures coincide with the signatures of the
+      respective methods in [Ast_traverse.map_with_expansion_context]. *)
+  val register_transformation
+  :  ?extensions       : Extension.t list (* deprecated, use ~rules instead *)
+  -> ?rules            : Context_free.Rule.t list
+  -> ?enclose_impl     : (Location.t option -> structure * structure)
```

Comment:
Sorry to think about it just now but do you know if it would make sense to pass the expansion_context here as well?

Reply:
Yes, I think you're right. That way we pass the expansion_context to all direct callbacks here. I've done that now.